terraform/deployments/cluster-infrastructure/aws_lb_controller_iam.tf

```diff
@@ -0,0 +1,263 @@
+# IAM role and policy to allow the AWS Load Balancer Controller to talk to AWS
```

Comment:
General question on structure - should we define everything as local TF modules? I personally quite like that approach as it means [you can see the high-level relationships](https://github.com/ministryofjustice/analytics-platform-ops/blob/main/infra/terraform/platform-base/main.tf) between logical components within the main.tf file. It does mean more work as you have to define more variables and outputs, but that does also mean you end up with something like an actual defined interface for each component.

Reply:
Hrm that's an interesting one. I do see your point of view, in fact I started out with that rule of thumb on the ECS project (which was at least an order of magnitude more complex so definitely some use of submodules) but I've since changed my mind based on that experience. My advice would be to use submodules only where TF code actually needs to be reused, because the verbosity and complexity that the module calls introduce is very expensive to maintain and really hurts readability.

I'd love to make things more modular and testable in principle, but the HCL/Terraform language imposes such a high maintenance penalty on doing that that I've come to the conclusion that it's only worth it where reuse is needed, rather than for structure. Thoughtful use of separate files within the same module can achieve most of the same benefits of surfacing the structure and responsibilities without the maintenance penalty.

Reply:
How strongly do you feel about this one? 😅

Reply:
Not that strongly really, certainly not at this stage - this just comes from personal experience of having a fairly large terraform project where it became really hard to see what was going on because so much of terraform's behaviour is implicit, and moving to an explicit input/output structure with modules really helped introduce some sanity. But, we're a long way away from that scenario at the moment, plus I don't want to impose my opinions on everyone - this kind of thing is a team decision in my mind.

Reply:
Totally happy to keep an eye on how this goes and refactor if it starts getting hairy again.

Maybe one day if I get a bunch of spare time I'll write a Dhall frontend for Pulumi 😁

Reply:
...only thing I'd add is that in the absence of locally defined modules we should probably all be clear on how we structure things - I'd vote for exactly what you've got here, all resources for a given logical component grouped together in either a single file or multiple files with a common prefix (`aws_lb_controller_iam.tf`, `aws_lb_controller_ec2.tf` if there was EC2 stuff for example), rather than a resource type-based structure like `iam.tf`, `s3.tf`, where any notion of logical components goes right out the window.

Reply:
💯